### DISCUSSION TITLE
elementary OS development strategy

### DISCUSSION THREAD

**Discussion by Letterus** [elementary/os#611] [Ideas]
Hello everyone,

I wanted to give some feedback to my usage of elementary OS. This may belong to another place like Reddit, but I don't like to register there, so I just post here. You may just copy and delete this if it does not fit.

What am I writing about: I've been using elementary OS for about half a year on my development machine, but stopped using it as a my daily driver. You may be interested into why.

First I greatly like the general ideas of elementary OS and I am a huge fan of the project. I like the overall design principles and the clear focus on a cohesive user experience. So I keep supporting the project even though I fail at using the OS currently.

What do I consider the main problem of the project's strategy? I think the project has too ambitious goals for such small (wo-)manpower to deliver the expected results. I think it is quite impressive what the project has achieved so far. But I think it is way too much work to develop basic apps and an "operating system" (meaning a GUI integrating with Ubuntu) at the same time. Looking at my daily usage at the end I found my self using only applications from the Ubuntu repos and literally none of the elementary apps, because none of those worked for me.

It still was a great experience for me, because LibreOffice, Evolution, QOwnNotes and VSCodium looked great and I could use them quite well (so the Gtk/Qt and icon styles really are great) - but it would have been even nicer if I had anything like a global menu that would integrate the menu of these apps even better (without wasting additional space).

I finally stopped using elementary OS because even the core features of the OS failed too often for me (and I fail to look up and link all the issues that would need to be addressed here, so I just list them for now):

1. AppCenter crashed at almost every update - I finally started doing updates using apt on CLI
2. Flatpak apps did not behave as I expected (access to common data or services like EDS was missing), so I installed everything using apt and Ubuntu sources
3. Kernel updates lead to an overflowing boot partition quite soon so I had to delete old kernels manually all the time
4. Basic features of the switchboard like adding and deleting printers were to buggy and failed to do what I expect
5. Online accounts switchboard only could add few of my accounts so I finally used Evolution to add what I needed
6. In comparison to Xfce the whole GUI (window manager?) seems to work quite slowly and does show delay on really some tasks
7. the window manager hung on certain tasks using much load - so the whole interface became completely unresponsive (this was the final one on which I gave up)
8. Missing a global menu is acceptable for common apps like LibreOffice and Evolution (even though it's a design flaw regarding the loss of space imo), but not having an app indicator still is a big thing missing. You get it with every DE, and everyone using Nextcloud and Seafile (over here: all the FLOSS people) needs it imho.
9. I was missing an upgrade path for the OS (how to use it in production without?!) - it's great you are working on this!

What worked really well: Everything looks quite well and basic important tasks like switching windows, apps and workspaces are great and easy to use. Especially the support of multiple displays was very good. Only GNOME and Wayland succeed to be as good here (Xfce and KDE don't).

So this is what I experienced. What would I like to see for futures strategic decisions?

1. Work more closely on core OS features to improve overall technical quality and leave app development to third parties.
2. Support third party devs even more to work on basic apps necessary to provide the core features desirable for basic tasks.

The former means to achieve a technical quality that fits the visual appearance of the OS and to not leave one quality drop below the other (this once was the greatest feature of a well known company out there which elementary OS does not yet meet - to my understanding).

The latter may be achieved to further push the app store concept, even though I personally do not like the decision to use Flatpak very much, but I'm fine with that.

So far, my 2 cent. I hope you're fine with me dropping it here.

All the best and keep up the good work
Johannes

**Comment by Letterus** [comment]
Pinging @danrabbit and @cassidyjames to let them know.

**Comment by GeneralAccess** [comment]
@Letterus 
> "I wanted to give some feedback to my usage of elementary OS."

I strongly support Flatpak. Apart from that I agree with everything you say. I tried elementaryOS a few years ago and was very impressed with its style, but found it wasn't mature enough for me to use. I tried 6.1 in November 2023 and found much the same - I immediately had problems with important stuff like the Software Installer and something else. I agree with your suggestion that more emphasis making sure the basics work would be a wise move. 
The look is something other OSs could aspire to :).

**Comment by Letterus** [comment]
I'm returning to elementary OS for one of my laptops. I'm checking the issues above and I'm trying to solve them. Since numbers 6 and 7 were the most severe ones I'm happy it looks as I found a solution. The issues seem to derive from an graphics driver issue on Lenovo Thinkpads of the X series (X250 to x280 at least). Solution: https://github.com/elementary/gala/issues/2023
